R/zchunk_LA1321.cement.R

```diff
@@ -52,72 +63,369 @@ module_energy_LA1321.cement_DISABLED <- function(command, ...) {
     L132.in_EJ_R_indenergy_F_Yh <- get_data(all_data, "L132.in_EJ_R_indenergy_F_Yh")
 
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+    # 2. Perform computations
+
+    # Set constants used for this chunk
+    # ---------------------------------
+
+    # Extract carbon coefficient for limestone from assumption file
+    LIMESTONE_CCOEF <- A_PrimaryFuelCCoef$PrimaryFuelCO2Coef[A_PrimaryFuelCCoef$PrimaryFuelCO2Coef.name == "limestone"]
+    # Determine historical years not available in data set (additional years) to be copy values from final available year (final_CO2_year)
+    ADDITIONAL_YEARS <- HISTORICAL_YEARS[!HISTORICAL_YEARS %in% energy.CDIAC_CO2_HISTORICAL_YEARS]
+    FINAL_CO2_YEAR <- energy.CDIAC_CO2_HISTORICAL_YEARS[length(energy.CDIAC_CO2_HISTORICAL_YEARS)]
+    # Sets maximum for IO coefficient calculated below
+    MAX_IOELEC <- 4
```

Comment:
Should this be moved to the constants file?

Reply:
@bpbond @kvcalvin I didnt move additional or final CO2 year because they're dependent on a time constant that would change under timeshift conditions. I was curious so I looked at some of the other sector constants and they do call time constants before they would be shifted, e.g. `energy.CDIAC_CO2_HISTORICAL_YEARS <- HISTORICAL_YEARS[HISTORICAL_YEARS < 2010]` and `emissions.MODEL_BASE_YEARS <- BASE_YEARS[BASE_YEARS < 2008]`. That would put unshifted values in those constants, right? Should I move the commented out timeshift code up to directly below the Time Constants section, so it would update these?

Reply:
>Should I move the commented out timeshift code up to directly below the Time Constants section, so it would update these?

Oh jeez good catch! But NO...let's not make trouble at this stage in the game. Leave the commented out timeshift code alone for now please.

Reply:
I started to open an issue for this, but then decided against it. We've always known the timeshift test was highly imperfect; it's meant to be a way to identify fragile code, but it's not exhaustive.

Reply:
Sounds good, I'll leave it well enough alone for now then.